### ISSUE TITLE
CLI not working for video ids starting with hyphen

### ISSUE THREAD

**Issue by ojoven** [jdepoix/youtube-transcript-api#80]
Hi, first thanks so much for this project, it's awesome!

I wanted to share a concern, I'm trying to retrieve the captions from this video with ID "-tiUDjn7-as" and I can't cause the hyphen in the start conflicts with the CLI parameters formatting:
```
youtube_transcript_api -tiUDjn7-as --languages en --exclude-generated --json
youtube_transcript_api: error: too few arguments
```

I thought that using quotes around the ID would work but still:
```
youtube_transcript_api "-tiUDjn7-as" --languages en --exclude-generated --json
youtube_transcript_api: error: too few arguments
```

Is there anything I may be missing?
I'm afraid I'm not very experienced on Python and that's why I'm asking but if some guidance is provided I may be able to do the necessary adjustment myself.

Thanks so much,
Mikel

**Comment by jdepoix** [comment]
Hi @ojoven,
thanks for reporting this. I never thought about this.

Unfortunately I am very busy right now and won't be able to fix this in the coming weeks, but I would appreciate you stepping in very much! You can find the CLI code under `youtube_transcript_api/_cli.py`. The arguments are parsed using [pythons argparse module](https://docs.python.org/3/library/argparse.html) and the argument you'll have to fix is called `video_ids`. I don't know how to fix this exactly though, so you'd have to do some further research.

However, if you just want to get your stuffing working, you could just write a python script and use this package as a dependency. It's pretty straight forward and you should be able to do it without much knowledge about python. [Check out this example](https://github.com/jdepoix/youtube-transcript-api#by-example). Your problem won't occur while passing in the video id in code, as this error is only related to the CLI input being parsed incorrectly.

**Comment by ojoven** [comment]
Hey, thanks so much for the detailed response. I'll try to fix this on my end and if I find a solution that can be shared I'll create a PR and add a new message to this thread, thanks.

**Comment by jdepoix** [comment]
Actually, I'll keep this issue open, as it is a bug which definitely should be fixed at some point.

**Comment by crhowell** [comment]
Hey yall, I took some time this morning to have a look at this issue. From what I can tell it seems the call to `argparse.parse_args()` is the one complaining because it immediately identifies any arguments (video_ids arguments) with a leading `-` as an option instead of an argument. Which is why it thinks you passed no arguments at all for your example use-case above.

This makes for an interesting issue since now you cant just directly forward YouTube video ids into the `.parse_args()` method as arguments. The error itself is raised from [_cli.py L134](https://github.com/jdepoix/youtube-transcript-api/blob/master/youtube_transcript_api/_cli.py#L134). 

I have an idea for a solution but I can't say I love it and I dont think it perfectly solves every case a video_id can be since `-` are valid characters for a YouTube video id and I am not sure there is anything stopping YouTube from prefixing a video id with `--` which is really a conflict between what we to do in the CLI versus what `argparse` needs to identify options/arguments.

**Comment by jdepoix** [comment]
Fixed and released in `v0.4.0`! Thanks @crhowell!
